src/mjolnir/ingest_transit.cc

```diff
@@ -0,0 +1,778 @@
+        }
+        tile_map[currId] = currTile;
+      }
+    }
+  }
+  std::priority_queue<tileTransitInfo> prioritized;
+  for (auto it = tile_map.begin(); it != tile_map.end(); it++) {
+    prioritized.push(it->second);
+  }
+  return prioritized;
+}
+
+std::unordered_map<gtfs::Id, GraphId> write_stops(Transit& tile, const tileTransitInfo& tile_info) {
+
+  const auto& tile_stopIds = tile_info.tile_stops;
+  auto node_id = tile_info.graphId;
+  feedCache stopFeeds;
+
+  std::unordered_map<gtfs::Id, GraphId> stop_graphIds;
+  // loop through all stops inside the tile
+  for (const feedObject& feed_stop : tile_stopIds) {
+    const auto& tile_stopId = feed_stop.id;
+    const auto& feed = stopFeeds(feed_stop);
+    auto* node = tile.add_nodes();
+    gtfs::Stop tile_stop = *(feed.get_stop(tile_stopId));
+    node->set_lon(tile_stop.stop_lon);
+    node->set_lat(tile_stop.stop_lat);
+    node->set_type((int)tile_stop.location_type);
+    node->set_graphid(node_id);
+    node_id++;
+    // node->set_prev_graphid(); what is prev?
+    node->set_name(tile_stop.stop_name);
+    // node->set_osm_way_id(); where is this?
+    node->set_timezone(tile_stop.stop_timezone);
+    // 0 is No Info ; 1 is True ; 2 is False
+    bool wheelchair_accessible = (tile_stop.wheelchair_boarding == "1");
+    node->set_wheelchair_boarding(wheelchair_accessible);
+    // node->set_generated(feed_stop.generated); ?
+    node->set_onestop_id(tile_stop.stop_id);
+    stop_graphIds[tile_stopId] = node_id;
+  }
+  return stop_graphIds;
+}
+
+// read feed data per stop, given shape
+float add_stop_pair_shapes(const gtfs::Stop& stop_connect,
+                           const gtfs::Shape& trip_shape,
+                           const gtfs::StopTime& pointStopTime) {
+  // check which segment would belong to which tile
+  if (pointStopTime.shape_dist_traveled > 0) {
+    return pointStopTime.shape_dist_traveled;
+  }
+  float dist_traveled = 0;
+  float min_sq_distance = INFINITY;
+  PointLL stopPoint = PointLL(stop_connect.stop_lon, stop_connect.stop_lat);
+  projector_t project(stopPoint);
+  for (int segment = 0; segment < (int)trip_shape.size() - 1; segment++) {
+    auto currOrigin = trip_shape[segment];
+    auto currDest = trip_shape[segment + 1];
+    PointLL originPoint = PointLL(currOrigin.shape_pt_lon, currOrigin.shape_pt_lat);
+    PointLL destPoint = PointLL(currDest.shape_pt_lon, currDest.shape_pt_lat);
+
+    PointLL minPoint = project(originPoint, destPoint);
+    float sq_distance = project.approx.DistanceSquared(minPoint);
+    if (sq_distance < min_sq_distance) {
+      min_sq_distance = dist_traveled + originPoint.Distance(minPoint);
+    }
+    dist_traveled += originPoint.Distance(destPoint);
+  }
+  return dist_traveled;
+}
+
+struct unique_transit_t {
+  std::mutex lock;
+  std::unordered_map<std::string, size_t> trips;
+  std::unordered_map<std::string, size_t> block_ids;
+  std::unordered_map<std::string, size_t> lines;
+};
```

Comment:
nit: can we move this to the top of the file? I do like it better when I know structs are always declared at the top of the file

Reply:
👍